### DISCUSSION TITLE
Adding buttons

### DISCUSSION THREAD

**Discussion by dotPIX** [gicamm/homeassistant-comelit#39] [Ideas]
Hi Gicamm,
It would be nice and very useful to have the buttons (input components) exposed from the comelit hub.
I was thinking of using the wall switches connected through the 20046501 module as triggers for some zigbee lights and other.
Is it possible to use these components from the hub? Thank you..

![btn](https://user-images.githubusercontent.com/3974826/229354489-d0c836e0-4e04-4fe5-89c5-89b44e3f1390.png)
![btn2](https://user-images.githubusercontent.com/3974826/229354755-a75bf56f-ccc1-462a-81aa-17e8573bb76b.png)

**Comment by gicamm** [comment]
Hi dot,
It could be possible. I'll go for it.

**Reply by dotPIX** [reply]
Hi gicam, do you have any news on this? May I be of assistance?

**Reply by gicamm** [reply]
Hi, currently nothing new for this one. If you are good in programming you could add this feature, otherwise you have to wait :D. Anyway you can do the same stuff with 'scene'. It's already supported.

**Reply by dotPIX** [reply]
I can manage to code something with python. I've tried to modify your code but I can't find a way to target buttons.
What do you mean with "scene"? HA or Comelit HUB?

**Reply by gicamm** [reply]
Please look at this
[https://github.com/gicamm/homeassistant-comelit/wiki#open-the-door
](https://github.com/gicamm/homeassistant-comelit/wiki#open-the-door
)
you can create a comelit 'scenario'. HA recognizes it as a 'scene.'

**Reply by dotPIX** [reply]
I use them already for the gate and the alarm. They are very useful for integrating comelit elements on HA. I need the other way around: command an HA element with a comelit element.

**Reply by dotPIX** [reply]
Hi @gicamm, I noticed the "switch" entity in your add-on documentation. What is it for and how can I use it on SimpleProg app? Thank you

**Reply by gicamm** [reply]
A switch can be used, for instance, to control modules like the 6I5O or 9I8O. Personally, I’m utilizing a couple of outputs to control certain devices.

**Reply by dotPIX** [reply]
I have two 9I80s that control some lights. Do you know how I can configure it on SimpleProg to show the "switch" entity on HA? Thanks
